processing/rdns_handler.go

```diff
@@ -0,0 +1,106 @@
+package processing
+
+// DCSO FEVER
+// Copyright (c) 2019, DCSO GmbH
+
+import (
+	"net"
+	"sync"
+
+	"github.com/DCSO/fever/types"
+	"github.com/DCSO/fever/util"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/yl2chen/cidranger"
+)
+
+// RDNSHandler is a handler that enriches events with reverse DNS
+// information looked up on the sensor, for both source and destination
+// IP addresses.
+type RDNSHandler struct {
+	sync.Mutex
+	Logger            *log.Entry
+	HostNamer         *util.HostNamer
+	PrivateRanges     cidranger.Ranger
+	PrivateRangesOnly bool
+}
+
+// MakeRDNSHandler returns a new RDNSHandler, backed by the passed HostNamer.
+func MakeRDNSHandler(hn *util.HostNamer) *RDNSHandler {
+	rh := &RDNSHandler{
+		Logger: log.WithFields(log.Fields{
+			"domain": "rdns",
+		}),
+		PrivateRanges: cidranger.NewPCTrieRanger(),
+		HostNamer:     hn,
+	}
+	for _, cidr := range []string{
+		"10.0.0.0/8",
+		"172.16.0.0/12",
+		"192.168.0.0/16",
+	} {
+		_, block, _ := net.ParseCIDR(cidr)
```

Comment:
as net.ParseCIDR also supports IPv6 addresses these would be handled too and we should included tests for this case IMHO.

Reply:
At the moment we don't test the private IP case explicitly since there's no way to properly mock DNS replies when we use `net.LookupAddr()`. However, I will add IPv6 unique local addresses (fc00::/7) to this list.